lib/protoflow-blocks/src/blocks/sys/read_socket.rs

```diff
@@ -0,0 +1,221 @@
+extern crate std;
+
+use crate::prelude::{vec, Bytes, String};
+use crate::{StdioConfig, StdioError, StdioSystem, System};
+use protoflow_core::{
+    Block, BlockError, BlockResult, BlockRuntime, OutputPort, Port, PortResult, SystemBuilding,
+};
+use protoflow_derive::Block;
+use serde::{Deserialize, Serialize};
+use simple_mermaid::mermaid;
+use std::{
+    format,
+    io::Read,
+    net::{TcpListener, TcpStream},
+    sync::{Arc, Mutex, PoisonError},
+};
+use tracing::{error, info};
+
+/// A block that reads a proto object from a TCP port.
+///
+/// # Block Diagram
+#[doc = mermaid!("../../../doc/sys/read_socket.mmd")]
+///
+/// # Sequence Diagram
+#[doc = mermaid!("../../../doc/sys/read_socket.seq.mmd" framed)]
+///
+/// # Examples
+///
+/// ## Using the block in a system
+///
+/// ```rust
+/// # use protoflow_blocks::*;
+/// # fn main() {
+/// System::build(|s| {
+///     // TODO
+/// });
+/// # }
+/// ```
+///
+/// ## Running the block via the CLI
+///
+/// ```console
+/// $ protoflow execute ReadSocket connection=tcp://127.0.0.1:7077 buffer_size="1024"
+/// ```
+///
+#[derive(Block, Clone)]
+pub struct ReadSocket {
+    #[output]
+    pub output: OutputPort<Bytes>,
+    #[parameter]
+    pub config: ReadSocketConfig,
+    #[cfg(feature = "std")]
+    #[state]
+    pub listener: Arc<Mutex<Option<TcpListener>>>,
+    #[cfg(feature = "std")]
```

Comment:
These two `#[cfg(feature = "std")]` are spurious and should be removed.

Reply:
removed